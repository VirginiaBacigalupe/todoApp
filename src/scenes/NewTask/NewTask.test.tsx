import { useRoute } from '@react-navigation/native'
import { shallow } from 'enzyme'
import React from 'react'
import { TextInput } from 'react-native'

import { TaskType } from '../../index'
import { NewTask } from './index'

jest.mock('@react-navigation/native')
jest.mock('react-redux')

const task: TaskType = {
  completed: false,
  description: 'desc tarea 1',
  id: '1',
  title: 'tarea1',
}

const mockUseRoute = useRoute as jest.Mock

afterEach(jest.clearAllMocks)

describe('NewTask', () => {
  it('render new task screen', () => {
    mockUseRoute.mockReturnValue({ params: {} })
    const wrapper = shallow(<NewTask />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(TextInput)).toHaveLength(2)
  })

  it('render edit screen', () => {
    mockUseRoute.mockReturnValue({ params: { task } })
    React.useState = jest
      .fn()
      .mockReturnValueOnce([task.description, {}])
      .mockReturnValueOnce([task.title, {}])

    const wrapper = shallow(<NewTask />)
    expect(wrapper).toMatchSnapshot()

    expect(wrapper.find(TextInput)).toHaveLength(2)

    const descTextInput = wrapper.find('[testID="inputDescription"]')
    descTextInput.simulate('change', { target: { value: task.description } })
    expect(descTextInput.prop('value')).toEqual(task.description)

    const titleTextInput = wrapper.find('[testID="inputTitle"]')
    titleTextInput.simulate('change', { target: { value: task.title } })
    expect(titleTextInput.prop('value')).toEqual(task.title)
  })
})
