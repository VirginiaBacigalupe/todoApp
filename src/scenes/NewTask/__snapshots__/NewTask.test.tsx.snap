// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTask render edit screen 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "space-around",
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Task title"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "rgb(255, 255, 255)",
        "borderBottomColor": "rgb(255, 25, 123)",
        "borderBottomWidth": 1,
        "fontSize": 20,
        "padding": "5%",
        "width": "100%",
      }
    }
    testID="inputTitle"
    underlineColorAndroid="transparent"
    value="tarea1"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Task description"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "rgb(255, 255, 255)",
        "padding": "5%",
        "paddingBottom": "20%",
        "width": "100%",
      }
    }
    testID="inputDescription"
    underlineColorAndroid="transparent"
    value="desc tarea 1"
  />
</KeyboardAvoidingView>
`;

exports[`NewTask render new task screen 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "space-around",
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Task title"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "rgb(255, 255, 255)",
        "borderBottomColor": "rgb(255, 25, 123)",
        "borderBottomWidth": 1,
        "fontSize": 20,
        "padding": "5%",
        "width": "100%",
      }
    }
    testID="inputTitle"
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Task description"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "rgb(255, 255, 255)",
        "padding": "5%",
        "paddingBottom": "20%",
        "width": "100%",
      }
    }
    testID="inputDescription"
    underlineColorAndroid="transparent"
    value=""
  />
</KeyboardAvoidingView>
`;
